pallets/ddc-nodes/src/migrations.rs

```diff
@@ -0,0 +1,194 @@
+use frame_support::{
+	storage_alias,
+	traits::{Get, GetStorageVersion, OnRuntimeUpgrade, StorageVersion},
+	weights::Weight,
+};
+use log::info;
+use serde::{Deserialize, Serialize};
+
+use super::*;
+use crate::{
+	storage_node::{StorageNode, StorageNodeProps},
+	ClusterId,
+};
+
+const LOG_TARGET: &str = "ddc-customers";
+
+pub mod v0 {
+	use frame_support::pallet_prelude::*;
+
+	use super::*;
+
+	// Define the old storage node structure
+	#[derive(Clone, Encode, Decode, RuntimeDebug, TypeInfo, PartialEq, Serialize, Deserialize)]
+	#[scale_info(skip_type_params(T))]
+	pub struct StorageNode<T: frame_system::Config> {
+		pub pub_key: StorageNodePubKey,
+		pub provider_id: <T as frame_system::Config>::AccountId,
+		pub cluster_id: Option<ClusterId>,
+		pub props: StorageNodeProps,
+	}
+
+	#[storage_alias]
+	pub type StorageNodes<T: Config> =
+		StorageMap<crate::Pallet<T>, Blake2_128Concat, StorageNodePubKey, StorageNode<T>>;
+}
+pub fn migrate_to_v1<T: Config>() -> Weight {
+	let on_chain_version = Pallet::<T>::on_chain_storage_version();
+	let current_version = Pallet::<T>::current_storage_version();
+
+	info!(
+		target: LOG_TARGET,
+		"Running migration with current storage version {:?} / onchain {:?}",
+		current_version,
+		on_chain_version
+	);
+
+	if on_chain_version == 0 && current_version == 1 {
+		let weight = T::DbWeight::get().reads(1);
+
+		let count = StorageNodes::<T>::iter().count();
+		info!(
+			target: LOG_TARGET,
+			" >>> Updating DDC Storage Nodes. Migrating {} nodes...", count
+		);
+		StorageNodes::<T>::translate::<v0::StorageNode<T>, _>(|_, old: v0::StorageNode<T>| {
+			info!(target: LOG_TARGET, "     Migrating node for node ID {:?}...", old.pub_key);
+
+			Some(StorageNode {
+				pub_key: old.pub_key,
+				provider_id: old.provider_id,
+				cluster_id: old.cluster_id,
+				props: old.props,
+				total_usage: None, // Default value for the new field
+			})
+		});
+
+		// Update storage version.
+		StorageVersion::new(1).put::<Pallet<T>>();
+		info!(
+			target: LOG_TARGET,
+			"Upgraded {} records, storage to version {:?}",
+			count,
+			current_version
+		);
+
+		weight.saturating_add(T::DbWeight::get().reads_writes(count as u64 + 1, count as u64 + 1))
+	} else {
+		info!(target: LOG_TARGET, " >>> Unused migration!");
+		T::DbWeight::get().reads(1)
+	}
+}
+
+pub struct MigrateToV1<T>(PhantomData<T>);
+
+impl<T: Config> OnRuntimeUpgrade for MigrateToV1<T> {
+	fn on_runtime_upgrade() -> Weight {
+		migrate_to_v1::<T>()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::DispatchError> {
+		let prev_count = StorageNodes::<T>::iter().count();
+
+		Ok((prev_count as u64).encode())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(prev_state: Vec<u8>) -> Result<(), sp_runtime::DispatchError> {
+		let prev_count: u64 =
+			Decode::decode(&mut &prev_state[..]).expect("pre_upgrade provides a valid state; qed");
+
+		let post_count = StorageNodes::<T>::iter().count() as u64;
```

Comment:
The purpose of the `post_upgrade` is to check that your migration applied the expected changes. You need to fetch the nodes here and check that every node has `total_usage == None`